lexers/asciidoc.lua

```diff
@@ -0,0 +1,110 @@
+-- Copyright 2025 Matěj Cepl. See LICENSE.
+-- Asciidoc LPeg lexer.
+
+local lexer = lexer
+local P, S, B = lpeg.P, lpeg.S, lpeg.B
+
+local lex = lexer.new(...)
+
+-- Distinguish between horizontal and vertical space so html start rule has a chance to match.
+lex:modify_rule('whitespace', lex:tag(lexer.WHITESPACE, S(' \t')^1 + S('\r\n')^1))
+
+-- Operators.
+lex:add_rule('operator', lex:tag(lexer.OPERATOR, S("*:^'`|~+<=>")))
+
+-- Keywords
+lex:add_rule('keyword', lex:tag(lexer.KEYWORD, lex:word_match(lexer.KEYWORD)))
+lex:set_word_list(lexer.KEYWORD, {"NOTE", "IMPORTANT", "WARNING", "TIP", "CAUTION", "TESTME"})
+
+-- Block elements.
+local function h(n)
+	return lex:tag(string.format('%s.h%s', lexer.HEADING, n),
+		lexer.to_eol(lexer.starts_line(string.rep('#', n))))
+end
+lex:add_rule('header', h(6) + h(5) + h(4) + h(3) + h(2) + h(1))
+
+lex:add_rule('hr',
+	lex:tag('hr', lpeg.Cmt(lexer.starts_line(lpeg.C(S('*-_')), true), function(input, index, c)
+		local line = input:match('[^\r\n]*', index):gsub('[ \t]', '')
+		if line:find('[^' .. c .. ']') or #line < 2 then return nil end
+		return (select(2, input:find('\r?\n', index)) or #input) + 1 -- include \n for eolfilled styles
+	end)))
+
+lex:add_rule('list', lex:tag(lexer.LIST,
+	lexer.starts_line(lexer.digit^1 * '.' + S('*+-'), true) * S(' \t')))
+
+local hspace = lexer.space - '\n'
+local blank_line = '\n' * hspace^0 * ('\n' + P(-1))
+
+local code_line = lexer.starts_line((B('    ') + B('\t')) * lexer.to_eol(), true)
+local code_block =
+	lexer.range(lexer.starts_line('```', true), '\n```' * hspace^0 * ('\n' + P(-1))) +
+		lexer.range(lexer.starts_line('~~~', true), '\n~~~' * hspace^0 * ('\n' + P(-1)))
+local code_inline = lpeg.Cmt(lpeg.C(P('`')^1), function(input, index, bt)
+	-- `foo`, ``foo``, ``foo`bar``, `foo``bar` are all allowed.
+	local _, e = input:find('[^`]' .. bt .. '%f[^`]', index)
+	return (e or #input) + 1
+end)
+lex:add_rule('block_code', lex:tag(lexer.CODE, code_line + code_block + code_inline))
+
+lex:add_rule('blockquote',
```

Comment:
This rule will not match since `>` is highlighted as an operator.

Reply:
see above